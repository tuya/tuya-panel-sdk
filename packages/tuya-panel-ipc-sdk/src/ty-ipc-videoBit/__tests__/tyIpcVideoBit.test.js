import React from 'react';
import { shallow } from 'enzyme';
import TYIpcVideoBit from '../index';

describe('TYIpcVideoBit components', () => {
  it('basic render', () => {
    const wrapper = shallow(
      <TYIpcVideoBit containerStyle={{ position: 'absolute', right: 0, top: 30 }} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('container render', () => {
    const wrapper = shallow(<TYIpcVideoBit valueStyle={{ color: 'red', fontSize: 24 }} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('bitTxtBox render', () => {
    const wrapper = shallow(<TYIpcVideoBit bitTxtBoxStyle={{ width: 100, height: 30 }} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('unit render', () => {
    const wrapper = shallow(<TYIpcVideoBit unitStyle={{ color: 'black' }} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('bit data', () => {
    const wrapper = shallow(<TYIpcVideoBit bitValue={30} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('unit data', () => {
    const wrapper = shallow(<TYIpcVideoBit unit="m/s" />);
    expect(wrapper).toMatchSnapshot();
  });
});