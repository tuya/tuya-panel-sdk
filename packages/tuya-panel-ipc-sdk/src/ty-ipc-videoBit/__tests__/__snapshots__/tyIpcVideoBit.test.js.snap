// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TYIpcVideoBit components basic render 1`] = `
<View
  style={
    Array [
      Object {
        "position": "absolute",
        "right": 0,
        "top": 20,
      },
      Object {
        "position": "absolute",
        "right": 0,
        "top": 30,
      },
    ]
  }
/>
`;

exports[`TYIpcVideoBit components bit data 1`] = `
<View
  style={
    Array [
      Object {
        "position": "absolute",
        "right": 0,
        "top": 20,
      },
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.3)",
          "borderRadius": 30,
          "height": 48,
          "justifyContent": "center",
          "width": 85,
        },
        Object {},
      ]
    }
  >
    <TYText
      align={null}
      color={null}
      size={null}
      style={
        Array [
          Object {
            "color": "#fff",
          },
          Object {},
        ]
      }
      type={null}
      weight={null}
    >
      30
       
      <TYText
        align={null}
        color={null}
        size={null}
        style={
          Array [
            Object {
              "color": "#fff",
            },
            Object {},
          ]
        }
        type={null}
        weight={null}
      >
        kb/s
      </TYText>
    </TYText>
  </View>
</View>
`;

exports[`TYIpcVideoBit components bitTxtBox render 1`] = `
<View
  style={
    Array [
      Object {
        "position": "absolute",
        "right": 0,
        "top": 20,
      },
      Object {},
    ]
  }
/>
`;

exports[`TYIpcVideoBit components container render 1`] = `
<View
  style={
    Array [
      Object {
        "position": "absolute",
        "right": 0,
        "top": 20,
      },
      Object {},
    ]
  }
/>
`;

exports[`TYIpcVideoBit components unit data 1`] = `
<View
  style={
    Array [
      Object {
        "position": "absolute",
        "right": 0,
        "top": 20,
      },
      Object {},
    ]
  }
/>
`;

exports[`TYIpcVideoBit components unit render 1`] = `
<View
  style={
    Array [
      Object {
        "position": "absolute",
        "right": 0,
        "top": 20,
      },
      Object {},
    ]
  }
/>
`;
